File: app/src/test/java/com/mgaetan89/showsrage/model/ShowsFilters_Status_GetStatusForStatesTest.java
package com.mgaetan89.showsrage.model;

import com.mgaetan89.showsrage.model.ShowsFilters.Status;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class ShowsFilters_Status_GetStatusForStatesTest {
	@Parameterized.Parameter(0)
	public boolean all;

	@Parameterized.Parameter(1)
	public boolean continuing;

	@Parameterized.Parameter(2)
	public boolean ended;

	@Parameterized.Parameter(4)
	public int status;

	@Parameterized.Parameter(3)
	public boolean unknown;

	@Test
	public void getStatusForStates() {
		assertThat(ShowsFilters.Status.Companion.getStatusForStates(this.all, this.continuing, this.ended, this.unknown)).isEqualTo(this.status);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		int all = Status.ALL.getStatus();
		int continuing = Status.CONTINUING.getStatus();
		int ended = Status.ENDED.getStatus();
		int unknown = Status.UNKNOWN.getStatus();

		return Arrays.asList(new Object[][]{
				{false, false, false, false, all},
				{false, false, false, true, unknown},
				{false, false, true, false, ended},
				{false, false, true, true, ended | unknown},
				{false, true, false, false, continuing},
				{false, true, false, true, continuing | unknown},
				{false, true, true, false, continuing | ended},
				{false, true, true, true, continuing | ended | unknown},
				{true, false, false, false, all},
				{true, false, false, true, all},
				{true, false, true, false, all},
				{true, false, true, true, all},
				{true, true, false, false, all},
				{true, true, false, true, all},
				{true, true, true, false, all},
				{true, true, true, true, all},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/model/ShowsFilters_Status_GetStatusForStatesTest.kt
package com.mgaetan89.showsrage.model

import com.mgaetan89.showsrage.model.ShowsFilters.Status
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ShowsFilters_Status_GetStatusForStatesTest(val all: Boolean, val continuing: Boolean, val ended: Boolean, val unknown: Boolean, val status: Int) {
    @Test
    fun getStatusForStates() {
        assertThat(ShowsFilters.Status.getStatusForStates(this.all, this.continuing, this.ended, this.unknown)).isEqualTo(this.status)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any>> {
            val all = Status.ALL.status
            val continuing = Status.CONTINUING.status
            val ended = Status.ENDED.status
            val unknown = Status.UNKNOWN.status

            return listOf(
                    arrayOf(false, false, false, false, all),
                    arrayOf(false, false, false, true, unknown),
                    arrayOf(false, false, true, false, ended),
                    arrayOf(false, false, true, true, ended or unknown),
                    arrayOf(false, true, false, false, continuing),
                    arrayOf(false, true, false, true, continuing or unknown),
                    arrayOf(false, true, true, false, continuing or ended),
                    arrayOf(false, true, true, true, continuing or ended or unknown),
                    arrayOf(true, false, false, false, all),
                    arrayOf(true, false, false, true, all),
                    arrayOf(true, false, true, false, all),
                    arrayOf(true, false, true, true, all),
                    arrayOf(true, true, false, false, all),
                    arrayOf(true, true, false, true, all),
                    arrayOf(true, true, true, false, all),
                    arrayOf(true, true, true, true, all)
            )
        }
    }
}
